parser-typechecker/src/Unison/PrintError.hs

```diff
@@ -639,9 +639,10 @@ renderTypeError e env src = case e of
         showSourceMaybes src [(,Type1) <$> rangeForAnnotated site],
         "\n\n",
         "This is likely due to a recursive function using abilities where the\n",
-        "signature does not specify the abilities used.",
+        "signature does not specify the abilities used. It may also be from a\n",
+        "variable introduced by an ability match escaping its scope.",
```

Comment:
One minor thing. Anytime you're editing error messages, it's a good opportunity to switch away from using hard linebreaks - hard linebreaks result in weird rendering depending on the user's terminal width. It's a bit of longstanding tech debt that the dynamic wrapping is used inconsistently in our error messages.

You can follow other examples in this file which use the pretty-printing library to get it to wrap dynamically based on the width of the terminal, using `Pr.wrap`, `Pr.lines`, etc.

I wouldn't necessarily hold up a PR just for this, but it's an example of "leave the code a bit nicer than you found it". :)